packages/vite/src/node/optimizer/index.ts

```diff
@@ -679,7 +685,7 @@ export async function findKnownImports(
   config: ResolvedConfig,
   ssr: boolean,
 ): Promise<string[]> {
-  const deps = (await scanImports(config)).deps
+  const { deps } = await scanImports(config).result
```

Comment:
hmm, result.then  has `return deps`, wouldn't that still need 
```suggestion
  const deps = await scanImports(config).result
```

or `return {deps}` in then?

Reply:
`scanImports` returns 
```
{
  cancel: () => Promise<void>
  result: Promise<{
    deps: Record<string, string>
    missing: Record<string, string>
  }>
}
```
So we can await `result` and gets it deps, no? Maybe you are thinking about `discoverProjectDependencies`?